File: examples/trafficLight.c
//trafficLight.c
//loops through red, yellow, green like a traffic light
//Author : Zachary Igielman

//to run:
//gcc -Wall -o trafficLight trafficLight.c -lwiringPi
//sudo ./trafficLight

//Import the libraries necessary for changing the state of the LEDs
#include <stdio.h>
#include <wiringPi.h>
#include <mcp23017.h>

//Create a function that turns the LEDs on/off depending on what values are sent to it when the function is called
void light(int a, int b, int c)
{
    digitalWrite(113,a);
    digitalWrite(114,b);
    digitalWrite(115,c);
}

int main (void)
{
    int i;
    //Setup the library for controlling GPIOs
    wiringPiSetup() ;
    //Setup communication to the PiPiano chip
    mcp23017Setup(100, 0x20) ;
    //Setup the LED pins as outputs
    for (i = 113 ; i < 116 ; i++)
        pinMode(i, OUTPUT) ;
    //Turn the LEDs on/off in the 'traffic lights sequence' 
    for (;;)
    {
        light(1,0,0);
        delay(3000);
        light(1,1,0);
        delay(1000);
        light(0,0,1);
        delay(5000);
        light(0,1,0);
        delay(1000);
    }
    return 0 ;
}

File: examples/trafficLight.py
#!/usr/bin/python
# trafficLight.py
# loops through red, yellow, green like a traffic light
# Author : Zachary Igielman

# to run:
# sudo python trafficLight.py

# Import the libraries necessary for changing the state of the LEDs
import RPi.GPIO as GPIO
import time
from Adafruit_I2C import Adafruit_I2C
from Adafruit_MCP230xx import *

# Determine the version of Raspberry Pi so the chip on the PiPiano gets picked up 
myBus=""
if GPIO.RPI_REVISION == 1:
    myBus=0
else:
    myBus=1

# Initialise the chip on the PiPiano and specify how many pins it has
mcp = Adafruit_MCP230XX(busnum = myBus, address = 0x20, num_gpios = 16)

# Setup the LED pins as outputs
i=13
while i<16:
    mcp.config(i,0)
    i=i+1

# Create a function that turns the LEDs on/off depending on what values are sent to it when the function is called
def light(a,b,c):
    mcp.output(13,a)
    mcp.output(14,b)
    mcp.output(15,c)

# Turn the LEDs on/off in the 'traffic lights sequence' 
while True:
    light(1,0,0)
    time.sleep(3)
    light(1,1,0)
    time.sleep(1)
    light(0,0,1)
    time.sleep(5)
    light(0,1,0)
    time.sleep(1)
